### PR TITLE
Add L2-side sanity checks

### PR DESCRIPTION
**Pull request by telome**

### CODE REVIEW

File: deploy/L2TokenGatewaySpell.sol [hunk 1]
```diff
@@ -38,11 +41,25 @@ contract L2TokenGatewaySpell {
     function rely(address usr) external { l2Gateway.rely(usr); }
     function deny(address usr) external { l2Gateway.deny(usr); }
     function close() external { l2Gateway.close(); }
+
+    function registerToken(address l1Token, address l2Token) public { 
+        l2Gateway.registerToken(l1Token, l2Token);
+        AuthLike(l2Token).rely(address(l2Gateway));
+    }
```
Comment (sunbreak1211) [inline deploy/L2TokenGatewaySpell.sol]: I guess the idea of extracting this as its specific function with also external visibility is to being able to use it directly in a future spell, right? Shouldn't be better to keep this function as it was receiving arrays?

File: deploy/TokenGatewayInit.sol [hunk 2]
```diff
@@ -97,7 +102,12 @@ library TokenGatewayInit {
 
         l1GovRelay.relay({
             target:            l2GatewayInstance.spell,
-            targetData:        abi.encodeCall(L2TokenGatewaySpell.registerTokens, (cfg.l1Tokens, cfg.l2Tokens)),
+            targetData:        abi.encodeCall(L2TokenGatewaySpell.init, (
+                l1Gateway_,
+                l1Router.counterpartGateway(),
```
Comment (sunbreak1211) [inline deploy/TokenGatewayInit.sol:108]:
```suggestion
                cfg.counterpartGateway,
```

Comment (sunbreak1211) [reply deploy/TokenGatewayInit.sol:108]: Nvm

File: deploy/L2TokenGatewaySpell.sol (hunk 1 above)
Comment (telome) [reply deploy/L2TokenGatewaySpell.sol]: Yes exactly. 
Yeah I think it's indeed better to move the loop to `registerTokens`. Done.

**Review by sunbreak1211 — APPROVED** [review]
LGTM

**Review by sunbreak1211 — APPROVED** [review]

**Review by oldchili — APPROVED** [review]
LGTM

**Pull request merged**